pyscript.core/docs/README.md

```diff
@@ -0,0 +1,341 @@
+ * [How Scripts Work](#how-scripts-work) - how `<script type="...">` works
+ * [How Events Work](#how-events-work) - how `<button py-click="...">` works
+ * [XWorker](#xworker) - how `XWorker` class and its `xworker` reference work
+ * [Custom Scripts](#custom-scripts) - how *custom types* can be defined and used to enrich any core feature
+
+
+## Terminology
+
+This section goal is to avoid confusion around topics discussed in this document, describing each *term* as exhaustively as possible.
+
+<details>
+  <summary><strong>Interpreter</strong></summary>
+  <div>
+
+Also commonly referred as *runtime* or *engine*, we consider an **interpreter** any "_piece of software_" able to parse, understand, and ultimately execute, a *Programming Language* through this project.
+
+We also explicitly use that "_piece of software_" as the interpreter name it refers to:
+
+ * [pyodide](https://pyodide.org/en/stable/index.html) is the name of the interpreter that runs likely the most complete version of latest *Python*, enabling dozen official modules at run time, also offering a great *JS* integration in its core
+ * [micropython](https://micropython.org/) is the name of the interpreter that runs a small subset of the *Python* standard library and is optimized to run in constrained environments such as *Mobile* phones, or even *Desktop*, thanks to its tiny size and an extremely fast bootstrap
+ * [wasmoon](https://github.com/ceifa/wasmoon) is the name of the interpreter that runs *Lua* on the browser and that, among the previous two interpreters, is fully compatible with all core features
+ * [ruby-wasm-wasi](https://github.com/ruby/ruby.wasm) is the name of the (currently *experimental*) interpreter that adds *Ruby* to the list of programming languages currently supported
+
+While internally the name `interpreter` is used to declare the identity of the reference, this value *must* be specified as `<script type>` attribute:
+
+```html
+<script type="pyodide">
+    import sys
+    print(sys.version)
+</script>
+
+<script type="micropython">
+    import sys
+    print(sys.version)
+</script>
+
+<script type="wasmoon">
+    print(_VERSION)
+</script>
+
+<script type="ruby-wasm-wasi">
+    print "ruby #{ RUBY_VERSION }"
+</script>
+```
+
+ℹ️ - Please note we decided on purpose to not use the generic programming language name instead of its interpreter project name to avoid being too exclusive for alternative projects that would like to target that very same *PL* (i.e. note *pyodide* & *micropython* not using *python* indeed as interpreter name).
+
+Last, but not least, it is possible regardless to create any *custom type* using any available interpreter, like we already do with the `<script type="py">` and its `<py-script>` custom element counter-part, both defined through the *@pyscript/core* API. In this case, the `type` won't refer anymore to the *interpreter* but to the *custom type* created in the module's registry.
```

Comment:
That's correct because we provide only 4 interpreters that must be pre-bundle and made available through our artifacts. A custom type builds up on top of that interpreter so I think your change makes sense.